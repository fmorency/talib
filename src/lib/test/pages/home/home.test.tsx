import renderer from "react-test-renderer";
import { expect, test } from "vitest";

import Home from "lib/pages/home/index";

const toJson = (component: renderer.ReactTestRenderer) => {
  const result = component.toJSON();
  expect(result).toBeDefined();
  return result as renderer.ReactTestRendererJSON;
};

test("Home", async () => {
  const component = renderer.create(<Home />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
