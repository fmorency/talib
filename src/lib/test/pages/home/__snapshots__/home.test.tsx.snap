// Vitest Snapshot v1

exports[`Home 1`] = `
[
  <div
    className="css-1z0t2y2"
  />,
  <div
    className="css-19bewft"
  >
    <div
      className="css-10qbm2z"
    >
      <div
        className="css-1molr2b"
      >
        <div
          className="css-945zx8"
        >
          <div
            className="css-ipfnbh"
          >
            <div
              className="chakra-stack css-12t3lkn"
            >
              <p
                className="chakra-text css-1kf9ore"
              >
                0
                M
              </p>
              <p
                className="chakra-text css-1yl1k0u"
              >
                (
                0
                 tps)
              </p>
            </div>
            <p
              className="chakra-text css-jf0soh"
            >
              TRANSACTIONS PER SECOND
            </p>
          </div>
          <div
            className="css-ipfnbh"
          >
            <div
              className="chakra-stack css-12t3lkn"
            >
              <p
                className="chakra-text css-1kf9ore"
              >
                0
              </p>
              <p
                className="chakra-text css-1yl1k0u"
              />
            </div>
            <p
              className="chakra-text css-jf0soh"
            >
              BLOCK HEIGHT
            </p>
          </div>
        </div>
        <div
          className="css-a2u0o1"
        >
          <div
            className="css-ipfnbh"
          >
            <div
              className="chakra-stack css-12t3lkn"
            >
              <p
                className="chakra-text css-1kf9ore"
              >
                0
                 GH/s
              </p>
              <p
                className="chakra-text css-1yl1k0u"
              />
            </div>
            <p
              className="chakra-text css-jf0soh"
            >
              HASH RATE
            </p>
          </div>
          <div
            className="css-ipfnbh"
          >
            <div
              className="chakra-stack css-12t3lkn"
            >
              <p
                className="chakra-text css-1kf9ore"
              >
                0
              </p>
              <p
                className="chakra-text css-1yl1k0u"
              >
                0
              </p>
            </div>
            <p
              className="chakra-text css-jf0soh"
            >
              COMPUTER POWER USED / AVAILABLE
            </p>
          </div>
        </div>
      </div>
      <div
        className="css-11nh8xu"
      >
        <div
          className="css-w3nq7d"
        >
          <div
            className="css-ipfnbh"
          >
            <div
              className="chakra-stack css-12t3lkn"
            >
              <p
                className="chakra-text css-1kf9ore"
              >
                0
                /
                0
              </p>
              <p
                className="chakra-text css-1yl1k0u"
              />
            </div>
            <p
              className="chakra-text css-jf0soh"
            >
              ACTIVE NODES
            </p>
          </div>
        </div>
        <div
          className="css-115z842"
        >
          <div
            className="css-xbl5nf"
          >
            <img
              alt="map"
              className="chakra-image css-1phd9a0"
              src="/assets/map.png"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="css-za6arg"
  >
    <div
      className="css-imrbmu"
    >
      <h3
        className="chakra-heading css-1i80fgp"
      >
        Block List
      </h3>
      <div
        className="css-1j3ysxt"
      >
        <div>
          <div
            className="css-5pj2v0"
          >
            <div
              className="css-u35zva"
            >
              <table
                className="chakra-table css-qegi10"
                role="table"
              >
                <thead
                  className="css-0"
                >
                  <tr
                    className="css-0"
                    role="row"
                  >
                    <th
                      className="css-0"
                      colSpan={1}
                      role="columnheader"
                    >
                      BLOCK
                    </th>
                    <th
                      className="css-0"
                      colSpan={1}
                      role="columnheader"
                    >
                      TIME
                    </th>
                    <th
                      className="css-0"
                      colSpan={1}
                      role="columnheader"
                    >
                      NUMBER OF TXS
                    </th>
                  </tr>
                </thead>
                <tbody
                  className="css-0"
                  role="rowgroup"
                >
                  <tr
                    className="css-0"
                    role="row"
                  >
                    <td
                      className="css-0"
                      role="cell"
                    >
                      <div
                        className="css-u4p24i"
                      >
                        <a
                          className="chakra-link css-zamdaj"
                          href="/block/457676765"
                        >
                          #457676765
                        </a>
                        <button
                          aria-label="copy to clipboard"
                          className="chakra-button css-1j1x3vb"
                          onClick={[Function]}
                          type="button"
                        >
                          <svg
                            aria-hidden={true}
                            fill="currentColor"
                            focusable={false}
                            height="1em"
                            stroke="currentColor"
                            strokeWidth="0"
                            style={
                              {
                                "color": undefined,
                              }
                            }
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <g>
                              <path
                                d="M0 0h24v24H0z"
                                fill="none"
                              />
                              <path
                                d="M7 6V3a1 1 0 0 1 1-1h12a1 1 0 0 1 1 1v14a1 1 0 0 1-1 1h-3v3c0 .552-.45 1-1.007 1H4.007A1.001 1.001 0 0 1 3 21l.003-14c0-.552.45-1 1.007-1H7zM5.003 8L5 20h10V8H5.003zM9 6h8v10h2V4H9v2z"
                              />
                            </g>
                          </svg>
                        </button>
                      </div>
                    </td>
                    <td
                      className="css-0"
                      role="cell"
                    >
                      <span>
                        2hs 36min ago
                      </span>
                    </td>
                    <td
                      className="css-0"
                      role="cell"
                    >
                      <span>
                        9582305095533
                      </span>
                    </td>
                  </tr>
                  <tr
                    className="css-0"
                    role="row"
                  >
                    <td
                      className="css-0"
                      role="cell"
                    >
                      <div
                        className="css-u4p24i"
                      >
                        <a
                          className="chakra-link css-zamdaj"
                          href="/block/457676766"
                        >
                          #457676766
                        </a>
                        <button
                          aria-label="copy to clipboard"
                          className="chakra-button css-1j1x3vb"
                          onClick={[Function]}
                          type="button"
                        >
                          <svg
                            aria-hidden={true}
                            fill="currentColor"
                            focusable={false}
                            height="1em"
                            stroke="currentColor"
                            strokeWidth="0"
                            style={
                              {
                                "color": undefined,
                              }
                            }
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <g>
                              <path
                                d="M0 0h24v24H0z"
                                fill="none"
                              />
                              <path
                                d="M7 6V3a1 1 0 0 1 1-1h12a1 1 0 0 1 1 1v14a1 1 0 0 1-1 1h-3v3c0 .552-.45 1-1.007 1H4.007A1.001 1.001 0 0 1 3 21l.003-14c0-.552.45-1 1.007-1H7zM5.003 8L5 20h10V8H5.003zM9 6h8v10h2V4H9v2z"
                              />
                            </g>
                          </svg>
                        </button>
                      </div>
                    </td>
                    <td
                      className="css-0"
                      role="cell"
                    >
                      <span>
                        4hs 19mins ago
                      </span>
                    </td>
                    <td
                      className="css-0"
                      role="cell"
                    >
                      <span>
                        9588675095533
                      </span>
                    </td>
                  </tr>
                  <tr
                    className="css-0"
                    role="row"
                  >
                    <td
                      className="css-0"
                      role="cell"
                    >
                      <div
                        className="css-u4p24i"
                      >
                        <a
                          className="chakra-link css-zamdaj"
                          href="/block/457676767"
                        >
                          #457676767
                        </a>
                        <button
                          aria-label="copy to clipboard"
                          className="chakra-button css-1j1x3vb"
                          onClick={[Function]}
                          type="button"
                        >
                          <svg
                            aria-hidden={true}
                            fill="currentColor"
                            focusable={false}
                            height="1em"
                            stroke="currentColor"
                            strokeWidth="0"
                            style={
                              {
                                "color": undefined,
                              }
                            }
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <g>
                              <path
                                d="M0 0h24v24H0z"
                                fill="none"
                              />
                              <path
                                d="M7 6V3a1 1 0 0 1 1-1h12a1 1 0 0 1 1 1v14a1 1 0 0 1-1 1h-3v3c0 .552-.45 1-1.007 1H4.007A1.001 1.001 0 0 1 3 21l.003-14c0-.552.45-1 1.007-1H7zM5.003 8L5 20h10V8H5.003zM9 6h8v10h2V4H9v2z"
                              />
                            </g>
                          </svg>
                        </button>
                      </div>
                    </td>
                    <td
                      className="css-0"
                      role="cell"
                    >
                      <span>
                        1hs 54mins ago
                      </span>
                    </td>
                    <td
                      className="css-0"
                      role="cell"
                    >
                      <span>
                        9588655095533
                      </span>
                    </td>
                  </tr>
                  <tr
                    className="css-0"
                    role="row"
                  >
                    <td
                      className="css-0"
                      role="cell"
                    >
                      <div
                        className="css-u4p24i"
                      >
                        <a
                          className="chakra-link css-zamdaj"
                          href="/block/457676768"
                        >
                          #457676768
                        </a>
                        <button
                          aria-label="copy to clipboard"
                          className="chakra-button css-1j1x3vb"
                          onClick={[Function]}
                          type="button"
                        >
                          <svg
                            aria-hidden={true}
                            fill="currentColor"
                            focusable={false}
                            height="1em"
                            stroke="currentColor"
                            strokeWidth="0"
                            style={
                              {
                                "color": undefined,
                              }
                            }
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <g>
                              <path
                                d="M0 0h24v24H0z"
                                fill="none"
                              />
                              <path
                                d="M7 6V3a1 1 0 0 1 1-1h12a1 1 0 0 1 1 1v14a1 1 0 0 1-1 1h-3v3c0 .552-.45 1-1.007 1H4.007A1.001 1.001 0 0 1 3 21l.003-14c0-.552.45-1 1.007-1H7zM5.003 8L5 20h10V8H5.003zM9 6h8v10h2V4H9v2z"
                              />
                            </g>
                          </svg>
                        </button>
                      </div>
                    </td>
                    <td
                      className="css-0"
                      role="cell"
                    >
                      <span>
                        6hs 56mins ago
                      </span>
                    </td>
                    <td
                      className="css-0"
                      role="cell"
                    >
                      <span>
                        9582305095533
                      </span>
                    </td>
                  </tr>
                  <tr
                    className="css-0"
                    role="row"
                  >
                    <td
                      className="css-0"
                      role="cell"
                    >
                      <div
                        className="css-u4p24i"
                      >
                        <a
                          className="chakra-link css-zamdaj"
                          href="/block/457676769"
                        >
                          #457676769
                        </a>
                        <button
                          aria-label="copy to clipboard"
                          className="chakra-button css-1j1x3vb"
                          onClick={[Function]}
                          type="button"
                        >
                          <svg
                            aria-hidden={true}
                            fill="currentColor"
                            focusable={false}
                            height="1em"
                            stroke="currentColor"
                            strokeWidth="0"
                            style={
                              {
                                "color": undefined,
                              }
                            }
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <g>
                              <path
                                d="M0 0h24v24H0z"
                                fill="none"
                              />
                              <path
                                d="M7 6V3a1 1 0 0 1 1-1h12a1 1 0 0 1 1 1v14a1 1 0 0 1-1 1h-3v3c0 .552-.45 1-1.007 1H4.007A1.001 1.001 0 0 1 3 21l.003-14c0-.552.45-1 1.007-1H7zM5.003 8L5 20h10V8H5.003zM9 6h8v10h2V4H9v2z"
                              />
                            </g>
                          </svg>
                        </button>
                      </div>
                    </td>
                    <td
                      className="css-0"
                      role="cell"
                    >
                      <span>
                        9hs 16mins ago
                      </span>
                    </td>
                    <td
                      className="css-0"
                      role="cell"
                    >
                      <span>
                        9588676095533
                      </span>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div
              className="css-37qczb"
            >
              <div
                className="chakra-stack css-rlwq96"
              >
                <div
                  className="chakra-stack css-1ss2m3q"
                >
                  <button
                    className="chakra-button css-dsj5zd"
                    type="button"
                  >
                    &lt;&lt;
                  </button>
                  <button
                    className="chakra-button css-gi0s1o"
                    disabled={true}
                    type="button"
                  >
                    1
                  </button>
                  <button
                    className="chakra-button css-59095b"
                    onClick={[Function]}
                    type="button"
                  >
                    2
                  </button>
                  <p
                    className="chakra-text css-8t3u5b"
                  >
                    ...
                  </p>
                  <button
                    className="chakra-button css-59095b"
                    onClick={[Function]}
                    type="button"
                  >
                    5
                  </button>
                  <button
                    className="chakra-button css-59095b"
                    onClick={[Function]}
                    type="button"
                  >
                    &gt;&gt;
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
